core/src/main/java/com/csi4999/systems/PhysicsObject.java

```diff
@@ -0,0 +1,53 @@
+package com.csi4999.systems;
+
+import com.badlogic.gdx.math.Vector2;
+public class PhysicsObject {
+
+    Vector2 velocity;
+    Vector2 acceleration;
+    Vector2 position;
+
+
+    public PhysicsObject(float initial_vX, float initial_vY, float initial_aX, float initial_aY,
+                         float initial_pX, float initial_pY) {
+
+        this.velocity = new Vector2(initial_vX , initial_vY);
+        this.acceleration = new Vector2(initial_aX , initial_aY);
+        this.position = new Vector2(initial_pX, initial_pY);
+    }
+
+    public PhysicsObject(Vector2 velocity , Vector2 acceleration, Vector2 position) {
+
+        this.velocity = velocity;
+        this.acceleration = acceleration;
+        this.position = position;
+    }
+
+    public PhysicsObject() {    }
+    public void move(float dt) {
+
+        velocity = velocity.mulAdd(acceleration, dt);
+        position = position.mulAdd(velocity, dt);
+
+    }
+    public float getX () {
+        return position.x;
+    }
+
+    public void setX (float X) {
+        if (position.x != X) {
```

Comment:
this likely performs worse than just position.x = x since the address of position.x needs to be retrieved twice, and copying a 32 bit float is cheap. honestly i dont think the getters/setters will be used since classes that extend PhysicsObject can just use the vectors directly, so you could probably remove them if you want

Reply:
removed the conditionals 
left the methods though would rather have them and remove them later